src/reputation/tasks.py

```diff
@@ -126,16 +173,49 @@ def check_transaction_success(transaction_hash):
     queue=QUEUE_PURCHASES,
 )
 def check_deposits():
-    deposits = Deposit.objects.filter(paid_status=None)
+    # Sort by created date to ensure a malicious user doesn't attempt to take credit for
+    # a deposit made by another user. This is a temporary solution until we add signed messages
+    # to validate users own wallets.
+    deposits = Deposit.objects.filter(paid_status=None).order_by("created_date")
+
     for deposit in deposits.iterator():
-        amt = deposit.amount
+        # If a deposit is not resolved after 6 hours, mark it as failed
+        if deposit.created_date < datetime.now(pytz.UTC) - timedelta(
+            seconds=TRANSACTION_AGE_LIMIT
+        ):
+            deposit.set_paid_failed()
+            continue
+
+        deposit_previously_paid = Deposit.objects.filter(
```

Comment:
So these would be deposits with the same transaction hash? Does it make sense to require that column to be unique so we don't get duplicates as such in the first place?

Reply:
Yeah, that's what I was originally thinking but the db seems to have duplicates and I was trying to find the easiest way to get this up and running. Happy to clean up the db and add uniqueness constraints.